### DISCUSSION TITLE
miniPaint in IPFS?

### DISCUSSION THREAD

**Discussion by ghost** [viliusle/miniPaint#347] [Q&A] [answered]
Hi.

#### Idea
I'm thinking of adding miniPaint to ipfs, that way anyone can access it without necessarily having to pay for hosting.

#### what is ipfs?
IPFS is a distributed system for storing and accessing files, websites, applications, and data. 

#### How to add a page in ipfs?
Please, see this [docs.ipfs.tech/how-to/websites-on-ipfs/single-page-website](https://docs.ipfs.tech/how-to/websites-on-ipfs/single-page-website/)

#### why?
1. IPFS makes the internet immutable and resilient. 
2. The CID is unique to the content and thus if the content is changed then a new CID is generated. The data is resilient since it is hosted on multiple computers, making it difficult to be completely removed from the network
3. Anyone can access miniPaint without necessarily having to pay for hosting.
4. IPFS is a decentralized network where users make files available among each other. 
5. If a website uses IPFS, it is served by a 'swarm' of people, much like BitTorrent users do when a file is shared. 
6. The advantage of this system is that websites can become completely decentralized.

#### question/feedback
What do you all think of the idea?

**Comment by viliusle** [comment, marked as answer]
We can add it. I still don't know much about IPFS or how popular it is. I need research more.

**Reply by ghost** [reply]
Hi viliusle.

> We can add it. I still don't know much about IPFS or how popular it is. I need research more.

An idea... "we can release pre-release versions on IPFS. People test (pre-release) and if they want to use the official version on GitHub pages or any own host they can too."

*What do you all think of the idea?*
